resolwe/flow/managers/listener.py

```diff
@@ -412,18 +412,19 @@ def handle_finish(self, obj):
                                 # Copy permissions.
                                 copy_permissions(parent_data, d)
 
+                                # Copy collection.
+                                d.collection = parent_data.collection
```

Comment:
We could do this in `DataDependency.objects.create` in line `406` to avoid another query (and ES reindexing).